v2/AdminCore.Services/EventService.cs

```diff
@@ -147,23 +147,49 @@ public EventDto CreateEvent(int employeeId, EventDateDto dates, EventTypes event
         EventDates = new List<EventDate>(),
         LastModified = _dateService.GetCurrentDateTime()
       };
+      if (IsHalfDay(dates))
+      {
+        newEvent.EventDates.Add(_mapper.Map<EventDate>(dates));
+      }
+      else
+      {
+        SplitEventIfFallsOnAWeekend(newEvent, dates.EndDate, dates.StartDate);
+      }
 
-      SplitEventIfFallsOnAWeekend(newEvent, dates.EndDate, dates.StartDate);
-      DatabaseContext.EventRepository.Insert(newEvent);
-      DatabaseContext.SaveChanges();
-      return _mapper.Map<EventDto>(newEvent);
+      if (DoesEmployeeHaveEnoughHolidays(employeeId, newEvent))
+      {
+        DatabaseContext.EventRepository.Insert(newEvent);
+        DatabaseContext.SaveChanges();
+        return _mapper.Map<EventDto>(newEvent);
+      }
+      throw new Exception("Not enough holidays to book");
     }
 
-    public void UpdateEvent(EventDateDto eventDateDto)
+    public void UpdateEvent(int employeeId, EventDateDto eventDateDto)
     {
       var eventToUpdate = GetEventById(eventDateDto.EventId);
 
       if (eventToUpdate != null)
       {
         eventToUpdate.EventDates.Clear();
-        SplitEventIfFallsOnAWeekend(eventToUpdate, eventDateDto.EndDate, eventDateDto.StartDate);
-        eventToUpdate.LastModified = _dateService.GetCurrentDateTime();
-        DatabaseContext.SaveChanges();
+        if (IsHalfDay(eventDateDto))
+        {
+          eventToUpdate.EventDates.Add(_mapper.Map<EventDate>(eventDateDto));
+        }
+        else
+        {
+          SplitEventIfFallsOnAWeekend(eventToUpdate, eventDateDto.EndDate, eventDateDto.StartDate);
+        }
+
+        if (DoesEmployeeHaveEnoughHolidays(employeeId, eventToUpdate))
```

Comment:
Personal preference - but If(EmployeeHasEnoughHolidays) reads better than if(DoesEmployeeHaveEnoughHolidays).
```suggestion
        if (EmployeeHasEnoughHolidays(employeeId, eventToUpdate))
```